### PR TITLE
isolate logging resources in separate namespace

### PR DESCRIPTION
**Pull request by saravanan30erd**
**What this PR does / why we need it**:
Now all the resources(ES, fluentd, kibana) are creating in kube-system namespace which supposed to have only k8s components, so isolated these resources in separate namespace `logging`.

**Release note**:

```release-note
fluentd: isolate logging resources in separate namespace
```

### CODE REVIEW

**Comment by neolit123** [comment]
thanks for the PR
please change the release note to:
`fluend: isolate logging resources in separate namespace`

/kind feature
/remove-sig cluster-lifecycle
/sig instrumentation
(TODO: need to add labels for this one.)

**Comment by coffeepac** [comment]
I think this will break updates of the addon.  I am in favor of this change but we need to be careful/explicit and possibly build in some backwards compatibility somehow.

File: cluster/addons/fluentd-elasticsearch/create-logging-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+kind: Namespace
+apiVersion: v1
+metadata:
+ name: logging
```
Comment (krmayankk) [inline cluster/addons/fluentd-elasticsearch/create-logging-namespace.yaml]: kube-logging ?

**Comment by monotek** [comment]
@coffeepac 
Imho also Elasticsearch should go to the new namespace: * https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/es-statefulset.yaml
* https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/es-service.yaml

What about the all the stuff in fluentd-gcp?
https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/fluentd-gcp

@krmayankk
I would also keep the name "logging" as also non kube apps would use the apps in this namespace.

**Comment by coffeepac** [comment]
This is something I want to do but I need to sort the upgrade path.

**Comment by coffeepac** [comment]
@saravanan30erd its been three years!  if you want to rebase this I'll approve/lgtm it.  I probably should have originally, this addon is not intended for production usage.  

I'll give you a week or so to comment and say you want to do the work or push a commit.  after that I'll push a new PR using these commits.

/remove-lifecycle stale

**Comment by saravanan30erd** [comment]
@coffeepac  Rebased the PR.

**Comment by coffeepac** [comment]
/approve

@saravanan30erd can you bump the minor version for any kubernetes objects that are being modified?   not the image versions.  I think this is the es statefulset, the fluentdaemonset, the kibana deployment and the configmap.

**Comment by saravanan30erd** [comment]
@coffeepac  updated the versions for k8s objects.

**Comment by coffeepac** [comment]
looks good!  @monotek do you have any comments or objections on this?  I'm assuming the chart overrides the namespace already anyway.

**Review by monotek — APPROVED** [review]
/lgtm
/approve

File: cluster/addons/fluentd-elasticsearch/es-statefulset.yaml [hunk 2]
```diff
@@ -28,15 +28,15 @@ rules:
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  namespace: kube-system
+  namespace: logging
```
Comment (pacoxu) [inline cluster/addons/fluentd-elasticsearch/es-statefulset.yaml:31]: ClusterRoleBinding is not namespaced.
This line should be removed.

File: cluster/addons/fluentd-elasticsearch/es-statefulset.yaml [hunk 3]
```diff
@@ -48,23 +48,23 @@ apiVersion: apps/v1
...
     matchLabels:
       k8s-app: elasticsearch-logging
-      version: v7.4.3
+      version: v7.4.4
   template:
     metadata:
       labels:
         k8s-app: elasticsearch-logging
-        version: v7.4.3
+        version: v7.4.4
```
Comment (pacoxu) [inline cluster/addons/fluentd-elasticsearch/es-statefulset.yaml:67]: - image: quay.io/fluentd_elasticsearch/elasticsearch:v7.4.3
The image here is 7.4.3.
Why the version is changed to v7.4.4 in labels?

Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-statefulset.yaml:67]: the label version is for the entire statefulset.   I believe updating the namespace is sufficient to rev the patch version at least.  but the image isn't moving.  

these will get back in-sync with the next minor release of the image.

File: cluster/addons/fluentd-elasticsearch/es-statefulset.yaml (hunk 2 above)
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-statefulset.yaml:31]: good catch, thank you, please remove this line @saravanan30erd

Comment (saravanan30erd) [reply cluster/addons/fluentd-elasticsearch/es-statefulset.yaml]: @coffeepac Its done

**Review by monotek — APPROVED** [review]

**Comment by coffeepac** [comment]
@saravanan30erd I know what the process is now.  however, in order to get added to a milestone (it doesn't matter which one) I will also need to sort out a new home for this addon.  it really shouldn't be in k/k anymore.

**Comment by saravanan30erd** [comment]
> @saravanan30erd I know what the process is now. however, in order to get added to a milestone (it doesn't matter which one) I will also need to sort out a new home for this addon. it really shouldn't be in k/k anymore.

@coffeepac  this addon still in k/k or is it moved to somewhere?

**Pull request merged**